web/modules/custom/dbcdk_community/src/Form/ProfileEditForm.php

```diff
@@ -176,43 +212,82 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       'birthday',
       'description',
     ];
+    $profile_fields = $this->updateProfileFields($fields, $form_state);
 
-    // Go through each field and check if there is any differences from the
-    // original value and the new value. If so, we add it to an array of new
-    // data. - We do this to make sure we actually have to PUT any data to the
-    // Community Service and provide a different message to the editor when
-    // nothing was updated or an actual update happened.
+    // Update roles based on the submitted "roles" input.
+    $profile_roles = $this->updateProfileRoles($form_state->getValue('roles'));
```

Comment:
Updating profile fields and roles may cause exceptions to be thrown. I think here is the right place to catch them and show an error message to the user.